app/models/certificate.server.ts

```diff
@@ -4,18 +4,19 @@ import { prisma } from '~/db.server';
 
 export type { Certificate } from '@prisma/client';
 
-export function getIssuedCertificateByUsername(username: Certificate['username']) {
-  /**
-   * There might be multiple certificates in the db for the same user, let's get
-   * the most recent one that has been successfully issued
-   */
+export async function getCertificateStatusByUsername(username: Certificate['username']) {
   return prisma.certificate
     .findMany({
-      where: { username, status: 'issued' },
-      orderBy: { validFrom: 'desc' },
+      where: {
+        username,
+        OR: [{ status: 'issued' }, { status: 'pending' }, { status: 'failed' }],
+      },
+      orderBy: { id: 'desc' },
       take: 1,
     })
-    .then(([certificate]) => certificate);
+    .then(([certificate]) => {
+      return certificate || {};
```

Comment:
I don't think we should return an empty object here.  Let's solve this problem at the call site (e.g., synthesize an object if you need one in React).

Reply:
Synthesize?

Reply:
I can create a try and catch, and when it fails, it will instead return an empty object with `status=failed` to send the user to the certificate "request" page

Reply:
At the level of the DB model, just returned undefined or a `certificate`, and do this other stuff in the API layer above.

Reply:
I wrote the try and catch on the loader itself, not sure if that's what you meant...

Reply:
OK, you can undo this change then.